### PR TITLE
Add developer section in docs

### PR DESCRIPTION
**Pull request by SimonHeybrock**
I propose to move the dev docs out of the API reference, adding a new section on top.

If we think there are too many entries not, maybe the tutorials can be moved into the user guide or getting started?

### CODE REVIEW

File: docs/developer/index.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Developer Area
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/developer/index.rst:1]: I don't mind. Pick one.

File: docs/index.rst
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/index.rst:1]: I think Getting Started is more important than our (almost embarrassing) selection of tutorials?

File: docs/developer/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvaytet) [reply docs/developer/index.rst:1]: +1 for `Development`

File: docs/index.rst
[the remark this answers is not in the document]
Comment (jl-wynen) [reply docs/index.rst:1]: Works for me

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**